### PR TITLE
Support capability enforcement environment variable in C++ [run-systemtest]

### PR DESCRIPTION
**Pull request by vekterli**
@havardpe @bjorncs please review

Mirrors Java enforce/log-only/disable semantics, defaulting to enforce.

Also fixes an issue where connection auth context and capabilities
would not be set if a server socket was running in mixed-mode. This
is not a problem in practice since mixed-mode is inherently completely
insecure since it must accept plain-text clients, which implicitly
have all capabilities granted.

### CODE REVIEW

**Review by bjorncs — APPROVED** [review]

File: fnet/src/vespa/fnet/frt/require_capabilities.cpp [hunk 1]
```diff
@@ -15,15 +16,22 @@ FRT_RequireCapabilities::allow(FRT_RPCRequest& req) const noexcept
...
         std::string method_name(req.GetMethodName(), req.GetMethodNameLen());
-        LOGBT(warning, peer_spec, "Permission denied for RPC method '%s'. "
+        LOGBT(warning, peer_spec, "%sPermission denied for RPC method '%s'. "
                                   "Peer at %s with %s. Call requires %s, but peer has %s",
+              ((mode == CapabilityEnforcementMode::LogOnly) ? "(Dry-run only, not enforced): " : ""),
               method_name.c_str(), peer_spec.c_str(),
               to_string(auth_ctx.peer_credentials()).c_str(),
               _required_capabilities.to_string().c_str(),
               auth_ctx.capabilities().to_string().c_str());
+        return (mode == CapabilityEnforcementMode::Enforce) ? false : true;
```
Comment (havardpe) [inline fnet/src/vespa/fnet/frt/require_capabilities.cpp:35]: prefer logic expression over conditional mapping of bool to bool

File: vespalib/src/vespa/vespalib/net/tls/capability_env_config.cpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+}
+
+const char* to_string(CapabilityEnforcementMode mode) noexcept {
+    switch (mode) {
+    case CapabilityEnforcementMode::Enforce: return "Enforce";
+    case CapabilityEnforcementMode::LogOnly: return "LogOnly";
+    case CapabilityEnforcementMode::Disable: return "Disable";
+    default: abort();
```
Comment (havardpe) [inline vespalib/src/vespa/vespalib/net/tls/capability_env_config.cpp:37]: prefer abort outside switch to enable compile-time check for enum completeness

**Review by havardpe — APPROVED** [review]

File: fnet/src/vespa/fnet/frt/require_capabilities.cpp (hunk 1 above)
Comment (vekterli) [reply fnet/src/vespa/fnet/frt/require_capabilities.cpp]: Fixed

File: vespalib/src/vespa/vespalib/net/tls/capability_env_config.cpp (hunk 2 above)
Comment (vekterli) [reply vespalib/src/vespa/vespalib/net/tls/capability_env_config.cpp]: Fixed

**Pull request merged**